atef/reduce.py

```diff
@@ -13,6 +13,27 @@
 from .util import run_in_executor
 
 
+@dataclass
+class EnumValue:
+    """
+    Wrapper for enum value that can be compared to either the string or int
+    value of the enum.  Only accepts string enums.
+    """
+    int_value: int
+    str_value: str
+
+    def __eq__(self, __value: object) -> bool:
+        if isinstance(__value, int):
```

Comment:
`return __value == self.int_value or __value == self.str_value`?

Since we're trying to help out the user here with common mistakes, is there a chance `__value` comes in erroneously as like `"3"` instead of (int) `3` and we need to do a `int(__value) == self.int_value`?

Reply:
I was thinking about the case where some aberrant enum (`['Out', '2', 'YAG']`), where both int and str "2" would be valid (so both types would be valid).  Do we think this is a fringe enough edge case to ignore?

Reply:
The answer I've come to is: yes probably